api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -292,7 +292,7 @@ public void setUserProvider(Provider<User> userProvider) {
 
     org.pmiops.workbench.db.model.Cohort cohort =
         cohortDao.findOne(cohortId);
-    if (cohort == null) {
+    if (cohort == null || cohort.getWorkspaceId() != workspace.getWorkspaceId()) {
```

Comment:
I believe so -- if you requested a cohort under the wrong workspace, you'd still get it back. And yeah this is bad since I think it would let you access cohorts you don't actually have permission to use. 

I will add a test.

Reply:
Done.